### DISCUSSION TITLE
How to use Cuda on my Computer

### DISCUSSION THREAD

**Discussion by apollo000104** [mrdbourke/pytorch-deep-learning#617] [Q&A]
Hello, Everybody.
I have nvidia geForce GTX 750Ti on my computer.
I can see
![2023-08-29_23-56-02](https://github.com/mrdbourke/pytorch-deep-learning/assets/133643755/a40e0dae-4799-49ab-9726-ec3f72df8aa8)
like this, with "!nvidia-smi" command.
But the result of this prompt is "False"
\torch.cuda.is_available()\

Why? How can I overcome this problem?
help me please.🙏

**Comment by otrturn** [comment]
Have you installed torch with cuda? I.e. check whether during the installation you used someting like `pip install torch==2.0.1+cpu`. In this case you may have a GPU on your system, but torch without cuda-support.

**Reply by apollo000104** [reply]
yes, I can see torch==2.0.1+cpu after the command torch.__version__.
So How can I do for new installation of torch with cuda????

**Reply by otrturn** [reply]
Check the site https://pytorch.org/get-started/locally/. You may have to uninstall torch.

**Reply by apollo000104** [reply]
Got it. I will try it.
Thanks

**Comment by apollo000104** [comment]
Hello,Thanks very much for your help.

I have tried, but it is not working.
I have uninstalled torch and newly install torch cuda 118
But it is not still working

**Reply by otrturn** [reply]
Hello Apollo,

my workstation with an NVIDIA GPU has arrived today. So, I must go through the same ordeal. I will document everything and will send it to you. How does that sound?

Ciao

Ralf

**Comment by apollo000104** [comment]
I wonder If you do so.
very thanks

**Reply by otrturn** [reply]
Hello Apollo,

I sure will. It is not an easy task to get the GPU running with jupyter, believe me. Ubuntu was pre-installed, the GPU was connected and worked. Then I tried to install the CUDA toolkit and the cuDNN Library. The result was that the workstation crashed as it mixed up drivers of different versions. As soon as I have better news, I update you.

**Reply by mrdbourke** [reply]
I've had similar issues setting up a deep learning PC in the past. 

I've collected a bunch of notes here on how I got mine setup.

In short, I had to *remove* all existing NVIDIA drivers and reinstall them.

These notes are rough and not a 100% working guide but they've got resources that helped me: https://mrdbourke.notion.site/Installing-CUDA-on-a-new-GPU-175916438b814525b8b161ac0c787667?pvs=4

**Comment by apollo000104** [comment]
Ok, thanks for the update.

**Comment by apollo000104** [comment]
Awesome, thanks!

**Reply by otrturn** [reply]
Has the issue been solved? I have been out of business for a while.

**Comment by apollo000104** [comment]
Yes, everything is fine.
Thanks.

**Comment by apollo000104** [comment]
How are things going? doing well?
If you have any problem on your work, call me plz.
I will help you

**Comment by oraclesoon** [comment]
I am using Asus VivoBook X540UBR (date: 01/08/2020) which comes with entry level NVIDIA GeForce MX110.

2 years ago, I abandoned Windows 10 and re-installed the notebook with Linux Mint 21.2 Cinnamon.

To my surprise, Linux has properly installed and maintained NVIDIA drivers via
**$ sudo apt update**
**$ sudo apt dist-upgrade**

I don't need to explicitly install Cuda toolkit from https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=20.04&target_type=deb_local

I simply install Cuda toolkit as hinted by **apt** via
**$ sudo apt install nvidia-cuda-toolkit**

Then, my GPU environment is ready.


To check the Cuda version in Ubuntu:

**$ nvidia-smi**
```
Sat Mar 16 21:16:35 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.161.07             Driver Version: 535.161.07   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce MX110           Off | 00000000:01:00.0 Off |                  N/A |
| N/A   38C    P8              N/A / 200W |      6MiB /  2048MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
```

To check whether Cuda toolkit is installed:

**$ nvcc --version**
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Thu_Nov_18_09:45:30_PST_2021
Cuda compilation tools, release 11.5, V11.5.119
Build cuda_11.5.r11.5/compiler.30672275_0
```

I follow https://github.com/mrdbourke/pytorch-deep-learning/blob/main/SETUP.md#tk---2-getting-setup-locally-linux-version.

```
import torch
print(torch.cuda.is_available())
print(torch.__version__)
```
True
1.10.0
